Fix bugs, make train, eval, predict optional via flags, and export predictions as JSONL

phucdev opened the pull request:
This pull request
- fixes a few bugs:
  - extra `max_length1` parameter in the data collator -> Removed
  - `encoder_guidance_mask` was `None` leading to an error when calling `_expand_mask` -> Added a check
  - `LookupError` when `punkt` had not been downloaded before running the script -> Added line to download
  - The test data loader used the same sampler for the eval (validation) dataset, which randomly samples a subset of the predict dataset -> Replaced `self._eval_sampler` call with `SequentialSampler(test_dataset)`
- adds:
  - CLI argument for specifying pre-trained weights path to replace hard-coded path
  - the summarization training script from transformers==4.30.2 for training the baseline model
  - ability to optionally train/evaluate/predict for e-bart/train.py
  - export of predictions to JSONL (adds generated summary to original example)
  - a script to split a file from the FullyAugmentedDataset into a data file and a guidance file

clement-gillet commented:
Hi Phuc,

I'm afraid I cannot work on the paper till next Tuesday because we got a
very decisive sprint at work coming up this week... I checked and I cannot
find a repo that contains BLANC, METEOR and BERT... I will easily do this
next week. If you manage solving the EBART output issue by then it's great,
otherwise I will take care of it myself..

KR,

Clément Gillet.

On Tue, 24 Sept 2024 at 12:43, Phuc Tran Truong ***@***.***>
wrote:

> This pull request
>
>    - fixes a few bugs:
>       - extra max_length1 parameter in the data collator -> Removed
>       - encoder_guidance_mask was None leading to an error when calling
>       _expand_mask -> Added a check
>       - LookupError when punkt had not been downloaded before running the
>       script -> Added line to download
>       - The test data loader used the same sampler for the eval
>       (validation) dataset, which randomly samples a subset of the predict
>       dataset -> Replaced self._eval_sampler call with
>       SequentialSampler(test_dataset)
>    - adds:
>       - CLI argument for specifying pre-trained weights path to replace
>       hard-coded path
>       - the summarization training script from transformers==4.30.2 for
>       training the baseline model
>       - ability to optionally train/evaluate/predict for e-bart/train.py
>       - export of predictions to JSONL (adds generated summary to
>       original example)
>       - a script to split a file from the FullyAugmentedDataset into a
>       data file and a guidance file
>
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/clement-gillet/E-BART/pull/1
> Commit Summary
>
>    - 32a835b
>    <https://github.com/clement-gillet/E-BART/pull/1/commits/32a835b608049833f68cdbf3b5437d77e0a2c205>
>    Fix bug
>    - c2a9f24
>    <https://github.com/clement-gillet/E-BART/pull/1/commits/c2a9f24c87e4df0597476fd89c242ae6bde37931>
>    Add .gitignore
>    - b27855b
>    <https://github.com/clement-gillet/E-BART/pull/1/commits/b27855bbfd0c5bea100601eb332507f6e771ffb0>
>    Update README.md and requirements.txt
>    - bab05cb
>    <https://github.com/clement-gillet/E-BART/pull/1/commits/bab05cb40c852d5c42411bb8ddf0f88324b03763>
>    Add run_summarization.py for training a plain BART model for summarization
>    - 8897ef9
>    <https://github.com/clement-gillet/E-BART/pull/1/commits/8897ef9c2bc1b6098779e44fc6a98f76dd4bc0a0>
>    Replace hard-coded paths
>    - 48285fb
>    <https://github.com/clement-gillet/E-BART/pull/1/commits/48285fb08e3bc49b256959978dfdea3657d78d05>
>    Handle "guidance mask is None" bug
>    - 2db86c2
>    <https://github.com/clement-gillet/E-BART/pull/1/commits/2db86c2d050caf744b3b3b3348e1f4c82f739c5e>
>    Download "punkt" for nltk tokenization
>    - 14622fe
>    <https://github.com/clement-gillet/E-BART/pull/1/commits/14622fe143665aa913eaa788ead95ac3fa890531>
>    Use workaround for race condition in evaluate:
>    https://github.com/huggingface/evaluate/issues/382
>    - 6a9d361
>    <https://github.com/clement-gillet/E-BART/pull/1/commits/6a9d361d5b0fb935d00dfcfbdcfd96521f1b67fa>
>    Add split_files.py
>    - 727bc4a
>    <https://github.com/clement-gillet/E-BART/pull/1/commits/727bc4a9aa5f611992681bc9000f5d62e940145b>
>    Write predictions to JSONL file
>    - 847a74d
>    <https://github.com/clement-gillet/E-BART/pull/1/commits/847a74d0ae8807fce95a6a9ad88c784398e86c0c>
>    Make training, evaluation and prediction optional via training arguments
>    flags
>    - 29768a0
>    <https://github.com/clement-gillet/E-BART/pull/1/commits/29768a077b24d22a17331b4f375a5ac035a9cca4>
>    Use SequentialSampler for test (predict) set instead of loading eval
>    sampler that is a RandomSampler based on self.eval_dataset
>    - 4ff488c
>    <https://github.com/clement-gillet/E-BART/pull/1/commits/4ff488cb3e464c62a2bc8c3720fc07971183d0d5>
>    For prediction add generated summary to original example for JSONL export
>
> File Changes
>
> (16 files <https://github.com/clement-gillet/E-BART/pull/1/files>)
>
>    - *A* .gitignore
>    <https://github.com/clement-gillet/E-BART/pull/1/files#diff-bc37d034bad564583790a46f19d807abfe519c5671395fd494d8cce506c42947>
>    (162)
>    - *D* .idea/.gitignore
>    <https://github.com/clement-gillet/E-BART/pull/1/files#diff-21610973868a98feff98dd0460438a8a32cca1447bc8701537ec5048e0c5faeb>
>    (3)
>    - *D* .idea/E-BART.iml
>    <https://github.com/clement-gillet/E-BART/pull/1/files#diff-7b211906354c73bf81958409c8b3eb80e043959ffd4c7b7c56a3ba18ee78c552>
>    (15)
>    - *D* .idea/inspectionProfiles/Project_Default.xml
>    <https://github.com/clement-gillet/E-BART/pull/1/files#diff-bbf0d02ebcf6764cae92fa4045a2d6f15f8a479cd1aa2ab4d1fd435a3aa74bbd>
>    (23)
>    - *D* .idea/inspectionProfiles/profiles_settings.xml
>    <https://github.com/clement-gillet/E-BART/pull/1/files#diff-9fb34448452829ed3cd715b49c0f4494fe0b97a4aef2a5b23e12c486ed932273>
>    (6)
>    - *D* .idea/misc.xml
>    <https://github.com/clement-gillet/E-BART/pull/1/files#diff-7ea1cd7bd5d16299a0abd33128e8d0a8bc04146f6b00c07bcd3cc0aad813c229>
>    (4)
>    - *D* .idea/modules.xml
>    <https://github.com/clement-gillet/E-BART/pull/1/files#diff-0aae258f1732eac2acd4fcdd1af6d0737b4ec8d233136c3ba7c40b6d49aeda50>
>    (8)
>    - *D* .idea/vcs.xml
>    <https://github.com/clement-gillet/E-BART/pull/1/files#diff-a6dab70bd0e9ffc91d8419ad52225baff3a3e7c9d8e1724f7ed6d96b661d804a>
>    (6)
>    - *M* README.md
>    <https://github.com/clement-gillet/E-BART/pull/1/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5>
>    (104)
>    - *M* e_bart/data/data_collator.py
>    <https://github.com/clement-gillet/E-BART/pull/1/files#diff-6b66827a0a97e4560599904b3ce875ce4cf56e1bb3b197cba56282da4321fcf0>
>    (1)
>    - *M* e_bart/model/modeling_e_bart.py
>    <https://github.com/clement-gillet/E-BART/pull/1/files#diff-ad0960954b447a92f4b417968ef0b821c3694b0b027f0133f09ce3dca2692ffb>
>    (8)
>    - *M* e_bart/train.py
>    <https://github.com/clement-gillet/E-BART/pull/1/files#diff-715666072928b73001bd122f391494f6a5bd1efb5c56863c01506660779d21fc>
>    (246)
>    - *M* e_bart/trainer.py
>    <https://github.com/clement-gillet/E-BART/pull/1/files#diff-49399dab599d1e0c70a9d70e3dccfbc65e38c6d09d1a116113d4d9a18b3f0d2d>
>    (3)
>    - *R* requirements.txt
>    <https://github.com/clement-gillet/E-BART/pull/1/files#diff-4d7c51b1efe9043e44439a949dfd92e5827321b34082903477fd04876edb7552>
>    (3)
>    - *A* run_summarization.py
>    <https://github.com/clement-gillet/E-BART/pull/1/files#diff-e561f9320174fca5c5439d322a246a51ce4030ac2adc099fddce8717e13de55b>
>    (769)
>    - *A* split_files.py
>    <https://github.com/clement-gillet/E-BART/pull/1/files#diff-9cdb316bc7b6b9e8aa927f4e70da94a5abb65494e77c2fa8208012d78ada2805>
>    (34)
>
> Patch Links:
>
>    - https://github.com/clement-gillet/E-BART/pull/1.patch
>    - https://github.com/clement-gillet/E-BART/pull/1.diff
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/clement-gillet/E-BART/pull/1>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AHNNH3D5VIVTEQ3PKFV6Q3TZYE66RAVCNFSM6AAAAABOX76YFOVHI2DSMVQWIX3LMV43ASLTON2WKOZSGU2DKMBTGUZDCNQ>
> .
> You are receiving this because your review was requested.Message ID:
> ***@***.***>
>

phucdev commented:
No worries. I figured out the cause of the E-BART output issue and fixed it in the pull request.

This was the cause:
- The test data loader (for the prediction) used the same sampler for the eval (validation) dataset, which randomly samples a subset of the predict dataset. That is why there were only predictions for 300 examples.

The solution:
- I replaced `self._eval_sampler` call with `SequentialSampler(test_dataset)` in commit https://github.com/clement-gillet/E-BART/pull/1/commits/29768a077b24d22a17331b4f375a5ac035a9cca4

I also reran the prediction for both models in order to get the new JSONL output format and uploaded the files to the DFKI cloud (in the model zip archives).

If you are fine with the changes I made then we can merge this with the main branch.
